Update guides for new min version

Jcodeerd opened the pull request:
After creating your pull request, tick these boxes if they are applicable to you.

- [ ] I have tested my changes against the `review` branch (the latest developmental version), and this pull request is targeting that branch as a base
- [ ] I have tested my changes on Python 3.5.3 or higher

----

### Description



### Related issues (if applicable)

Jcodeerd committed:
Update raspbian.md

need to make:
- python 3.8 installer script

Jcodeerd committed:
Update raspbian.md

created installer, (credits to Ole)
added deps
changed python version
script works for raspbian on a pi 4, not yet on 3b+

Jcodeerd committed:
updated installing files

not done yet, the installers are still WIP and being tested

Jcodeerd committed:
Update centos.md

add sh for the 7.4 codeblock

Jcodeerd committed:
Update centos.md

update centOS to python 3.8

Jcodeerd committed:
Update installer_centOS_8.sh

delete unused lines

Jcodeerd committed:
Update updating.md

update the new minimal version to python 3.8.0

Jcodeerd committed:
Update updating.md

update new version of MusicBot

Jcodeerd committed:
Update debian.md

replace curl by wget, as there is no point of using  curl

Jcodeerd committed:
Change python 3.8.0 to 3.8.7 to be the same as the issue template

Jcodeerd committed:
Create android.md

Jcodeerd committed:
Create aws.md

Jcodeerd committed:
Create nssm.md

Jcodeerd committed:
Create pm2.md

Jcodeerd committed:
Update android.md

Jcodeerd committed:
remove double links

Jcodeerd committed:
Update nssm.md

Jcodeerd committed:
Update aws.md

Jcodeerd committed:
Update android.md

Jcodeerd committed:
Update android.md

Jcodeerd committed:
add intents

Jcodeerd committed:
change scripts location

changed the location to a `scripts/`  folder to keep it organised

Jcodeerd committed:
Update mac.md

AleanaA commented on _docs/installing/aws.md:
```diff
@@ -0,0 +1,43 @@
+
+Before you can run a program through AWS, you must first create an EC2 Instance. To do this, Navigate to the “Services” tab in the top left and select EC2.
+
+[![](https://aleanaazure.xyz/lib/exe/fetch.php?w=512&h=512&tok=887f2d&media=https%3A%2F%2Fbritishbenji.github.io%2FMusicBot%2Fimages%2FAWS%2FSelect_EC2.PNG)](https://aleanaazure.xyz/lib/exe/fetch.php?tok=a8e437&media=https%3A%2F%2Fbritishbenji.github.io%2FMusicBot%2Fimages%2FAWS%2FSelect_EC2.PNG "https://britishbenji.github.io/MusicBot/images/AWS/Select_EC2.PNG")
```
Images should be hosted in the `/images` folder. Keeping them hosted elsewhere can lead to dead links, as is the case with this one. This applies to all images listed in this file, as they all seem to be referencing back to the unofficial guide's now defunct URLs referencing back to @BritishBenji 's now dead AWS image links.

Jcodeerd replied:
I think I haven't changed that link or so, just line 29 in this file

AleanaA commented on _docs/installing/raspbian.md:
```diff
@@ -14,17 +14,28 @@ If you're willing to try it, you can run the following commands in order to inst
+./installer.sh
+
+
+
```
This is a lot of unnecessary white space. We should only have 1-2 spacing lines, not 4.

Jcodeerd replied:
Fixed

AleanaA commented on _docs/installing/raspbian.md:
```diff
@@ -14,17 +14,28 @@ If you're willing to try it, you can run the following commands in order to inst
 sudo apt-get update -y
 sudo apt-get upgrade -y
 
+# Installer for python 3.8
```
Although this solution for streamlining Py3.8 installation is good, we can do it one better by piping the script into our shell and exectuting it that way instead. For example:
`curl -s http://example.com/install.sh | sh` would download and execute the `install.sh` script from http://example.com
There's an argument to be made about transparency, but if a user is following a guide like this, odds are they aren't going to read the script anyways.
Regardless, uploading the script to the github page is a better bet than using a discord attachment URL, allowing for the user to review the script on github prior to following the guide, should they wish to.

AleanaA replied:
Along these lines, in line with the argument for transparency, I honestly think the commands to install python should be included in the guide rather than through a script.
Letting the user get a feel for what they're actually doing with their system will help us out a lot in the long run, and if we're going to make automated scripts like this, we should sooner make full bot installation scripts rather than automating the installation of a single dependency.

Jcodeerd replied:
Okay, I will "un-script" it and put it here.

AleanaA commented on _docs/installing/raspbian.md:
```diff
@@ -14,17 +14,28 @@ If you're willing to try it, you can run the following commands in order to inst
-sudo apt install libopus-dev
-sudo apt install ffmpeg
+sudo apt -y install python3-dev
+sudo apt -y install python3-pip
```
Given we install python 3.8 earlier, we shouldn't need to install python3-dev or python3-pip.
Realistically these are only likely to cause extra conflict, and should pip not be installed with the installer script above, can easily be installed with the get-pip.py script.

Jcodeerd replied:
Removed

AleanaA commented on _docs/installing/scripts/installer_centOS_8.sh:
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
```
Similar to my comments about the Rasbian guide, I'm doubtful scripts like this will do us much good in the long run without just automating the bot's installation process in it's entirety. Attempts have been done to do this and can be found in the review branches main folder, but have not been maintained.

Jcodeerd replied:
Removed the script and put it in the centos page

AleanaA commented on _docs/installing/ubuntu.md:
```diff
@@ -70,20 +75,20 @@ sudo add-apt-repository ppa:deadsnakes/ppa -y
+sudo python3.8 -m pip install -U -r requirements.txt 
 ~~~
 
 ## Ubuntu 14.04
```
We're at a point in time of 14.04's life cycle that we'd be better off removing 14.04 from the guides entirely to clear up space for 22.04 once that releases.

Jcodeerd replied:
Removed.

AleanaA commented on _docs/installing/windows.md:
```diff
@@ -3,19 +3,17 @@ title: Windows
 3. Install [Git for Windows](http://gitforwindows.org/).
 4. During the setup, tick `Git from the command line and also 3rd-party software`, `Checkout Windows-style, commit Unix-style endings`, and `Use MinTTY (the default terminal MSYS2)`.
 5. Open Git Bash by right-clicking an empty space inside of a folder (e.g My Documents) and clicking `Git Bash here`.
 6. Run `git clone https://github.com/Just-Some-Bots/MusicBot.git MusicBot -b master` in the command window that opens.
```
It's important to note that the master branch does not support 3.8+ currently.
Each guide that has 3.8 installed rather than 3.7 should be updated to use the review branch instead.

Jcodeerd replied:
Done.

AleanaA commented on _docs/using/intents.md:
```diff
@@ -0,0 +1,22 @@
+order: 8
+---
+
+#### How to enable Intents?
```
Although this is a good section, this section doesn't need it's own page, and can be consolidated into the Troubleshooting section of the wiki.

Jcodeerd replied:
Done.

AleanaA commented on _docs/using/updating.md:
```diff
@@ -8,7 +8,7 @@ order: 4
 [![Python version](https://img.shields.io/badge/python-3.5%2C%203.6%2C%203.7-blue.svg)](https://python.org)
 
-> MucicBot version 1.9.8 requires Python 3.5.3 or higher. If you are updating the MusicBot with version below than 1.9.8, reinstall Python with the following version.
+> MucicBot version 02.01.21 requires Python 3.8.0 or higher. If you are updating the MusicBot with version below than 02.01.21, reinstall Python with the following version.
```
Remember to update the links to Python to properly point to 3.8+, rather than pre-3.8 links.

Jcodeerd replied:
Changed to 3.8

AleanaA commented on _docs/using/configuration.md:
```diff
@@ -8,6 +8,11 @@ Configuring the bot is relatively straight forward. First, open your bot folder
 
 > Editing your configuration file using Notepad or WordPad will result in problems as they save files in strange formats and aren't designed for editing .ini files.
 
+#### Intents
```
This section can be moved to troubleshooting, along with the information about this issue.

Jcodeerd replied:
Done.

AleanaA commented on _docs/installing/nssm.md:
```diff
@@ -0,0 +1,17 @@
+# NSSM
```
Because NSSM and PM2 are more along the lines of launching and keeping the bot running, I feel like these could go more in pages under Using the Bot

Jcodeerd replied:
Moved.

AleanaA commented on _docs/installing/debian.md:
```diff
@@ -3,31 +3,35 @@ title: Debian
 # Install dependencies
 sudo apt-get install git libopus-dev libffi-dev libsodium-dev ffmpeg -y
-sudo apt-get install build-essential libncursesw5-dev libgdbm-dev libc6-dev zlib1g-dev libsqlite3-dev tk-dev libssl-dev openssl -y
+sudo apt-get install build-essential libncursesw5-dev libgdbm-dev libc6-dev zlib1g-dev libsqlite3-dev tk-dev libssl-dev openssl libncurses5-dev libnss3-dev libreadline-dev wget libbz2-dev -y
```
Double check if Curl comes with Debian, and verify if all of these dependencies are necessary. Notably libncurses5-dev and libncursesw5-dev likely share libraries, and we likely don't need both. We shouldn't unnecessarily install new dependencies on a users system and add bloat when it's not needed.

Jcodeerd replied:
Curl added incase. Removed libreadline-dev.

Jcodeerd replied:
Wait a sec, why do we need curl actually? This page only uses wget.

AleanaA replied:
> Double check if Curl comes with Debian

I'm more so meaning we shouldn't install wget on a users system if the system already has Curl.

AleanaA reviewed (changes requested):
Overall, there's good changes, but there are a number of glaring issues that hold me back from approving this for merge. Especially with Master becoming a stale branch, it's best if we guide people on using Review rather than Master if we want to continue moving forward towards using Python 3.8

Jcodeerd committed:
Change / remove after feedback.

Change / remove after feedback.

AleanaA reviewed (changes requested):
Remember for the several files that lack it, to add the descriptor header as seen in every wiki page. This helps ensure the wiki knows where to place the guide in the list.

Jcodeerd committed:
Fix headers

Fixes some headers

Jcodeerd committed:
Update supported py versions

Jcodeerd committed:
fix 404

to make sure this doesnt collide with PR #2328

Jcodeerd committed:
update discord domain

Jcodeerd committed:
add headers

Jcodeerd committed:
Merge branch 'gh-pages' of https://github.com/Just-Some-Bots/MusicBot into patch-3

Jcodeerd committed:
update python version to 3.8.17

Jcodeerd committed:
update discord domain

Jcodeerd committed:
Fix missing bracket

Jcodeerd committed:
Update branches to pull from master instead of review

BabyBoySnow commented on _docs/using/intents.md:
```diff
@@ -0,0 +1,22 @@
+1. Go to Discord's [applications page](https://discord.com/developers/applications/me) (logging in if prompted).
+2. Select your bot.
+3. Go to the page called "Bot", and scroll until you see "Privileged Gateway Intents"
+4. Now enable "PRESENCE INTENT" and "SERVER MEMBERS INTENT" and you should be good to go!
```
Shouldn't we have them enable message content as well since we use on_message as a command handler?

Jcodeerd replied:
Good one, I'll add it aswell just to be safe

Jcodeerd committed:
remove sudo to install pip deps

Jcodeerd committed:
Update intents.md

Jcodeerd committed:
Update aws.md

fix order

Jcodeerd committed:
change python to 3.8.18 and get started on rocky guide

Jcodeerd committed:
Update rocky.md

Jcodeerd committed:
Update rocky.md

Jcodeerd committed:
update py and sync branches

Updated the python version as far as possible to 3.10.13, and made sure that all guides use master branch
